packages/components/src/components/date-picker/DatePicker.tsx

```diff
@@ -484,6 +491,16 @@ class DatePicker extends Component<
     }
   }
 
+  private handleOnEscAction(e) {
+    const { showPicker } = this.state;
+    if (showPicker) {
+      this.handleOnClose();
+    } else {
+      const element = (e.target as HTMLElement).parentNode;
+      element?.dispatchEvent(new KeyboardEvent(e.type, e));
+    }
+  }
+
```

Comment:
Could you explain the rationale behind this change ?

Reply:
After pressing Esc from the keyboard, if the datePicker is open, it'll close it. Otherwise, it triggers the Esc event and assigns it to the parrent element =>that is Advanced Search Modal. Refer to this bug https://perzoinc.atlassian.net/browse/C2-24820 => this is the reason to have above code

Reply:
I see your point, but this is not how you should fix it.

```suggestion
  private handleOnEscAction(e) {
    const { showPicker } = this.state;
    if (showPicker) {
      this.handleOnClose();
      e.stopPropagation();
    }
  }

```

Reply:
Hi @benoit-sy, I've already worked around the Esc function in Lite; it would be preferable if we moved it to ui-toolkit so that we wouldn't have to do it again if DatePicker is called in another location later.

Reply:
@NamNguyenFPT my point was that the fix/workaround in Lite should be removed as not needed at all, and the only fix to be done (that would fix the Lite issue we had 3 months ago) is my suggestion above.

when we hit ESC in the datepicker, and it is opened, we should close it and stop the propagation of the event, as it is already consumed by the datepicker, and we don't want any parent node to consume that event as it will bubble up the dom tree.